core/lib/src/rocket.rs

```diff
@@ -794,24 +806,52 @@ impl Rocket {
     /// # }
     /// ```
     // TODO.async Decide on an return type, possibly creating a discriminated union.
-    pub fn launch(self) -> Box<dyn std::error::Error> {
+    pub fn launch(self) -> Result<(), Box<dyn std::error::Error>> {
         // TODO.async What meaning should config.workers have now?
         // Initialize the tokio runtime
         let runtime = tokio::runtime::Builder::new()
             .core_threads(self.config.workers as usize)
             .build()
             .expect("Cannot build runtime!");
 
-        // TODO.async: Use with_graceful_shutdown, and let launch() return a Result<(), Error>
         match self.spawn_on(&runtime) {
-            Ok(fut) => match runtime.block_on(fut) {
-                Ok(_) => unreachable!("the call to `block_on` should block on success"),
-                Err(err) => err,
-            }
-            Err(err) => Box::new(err),
+            Ok(fut) => runtime.block_on(fut).map(|_| ()),
```

Comment:
This `.map(|_| ())` shouldn't be needed -- what is the type of this `_`?

Reply:
Looks like it wasn't needed. Removed.